modin/experimental/sql/hdk/query.py

```diff
@@ -0,0 +1,116 @@
+from modin.experimental.core.execution.native.implementations.hdk_on_native.hdk_worker import (
+    HdkWorker,
+)
+
+
+def hdk_query(query: str, **kwargs) -> pd.DataFrame:
+    """
+    Execute SQL queries on the HDK backend.
+
+    DataFrames are referenced in the query by names and are
+    passed to this function as name=value arguments.
+
+    Here is an example of a query to three data frames:
+
+    ids = [1, 2, 3]
+    first_names = ["James", "Peter", "Claus"]
+    last_names = ["Bond", "Pan", "Santa"]
+    courses_names = ["Mathematics", "Physics", "Geography"]
+    student = pd.DataFrame({"id": ids, "first_name": first_names, "last_name": last_names})
+    course = pd.DataFrame({"id": ids, "course_name": courses_names})
+    student_course = pd.DataFrame({"student_id": ids, "course_id": [3, 2, 1]})
+    query = '''
+    SELECT
+        student.first_name,
+        student.last_name,
+        course.course_name
+    FROM student
+    JOIN student_course
+    ON student.id = student_course.student_id
+    JOIN course
+    ON course.id = student_course.course_id
+    ORDER BY
+        last_name
+    '''
+    res = hdk_query(query, student=student, course=course, student_course=student_course)
+    print(res)
+
+    Parameters
+    ----------
+    query : str
+        SQL query to be executed.
+    **kwargs : **dict
+        DataFrames referenced by the query.
+
+    Returns
+    -------
+    modin.pandas.DataFrame
+        Execution result.
+    """
+    worker = HdkWorker()
+    if len(kwargs) > 0:
+        query = _build_query(query, kwargs, worker.import_arrow_table)
+    return from_arrow(worker.executeDML(query))
+
+
+def _build_query(query: str, frames: dict, import_table: callable) -> str:
+    """
+    Build query to be executed.
+
+    Table and column names are mapped to the real names
+    using the WITH statement.
+
+    Parameters
+    ----------
+    query : str
+        SQL query to be processed.
+    frames : dict
+        DataFrames referenced by the query.
+    import_table : callable
+        Used to import tables and assign the table names.
+
+    Returns
+    -------
+    str
+        SQL query to be executed.
+    """
+    alias = []
+    for name, df in frames.items():
+        assert isinstance(df._query_compiler, DFAlgQueryCompiler)
+        mf = df._query_compiler._modin_frame
+        if not mf._has_arrow_table():
+            mf._execute()
+        assert mf._has_arrow_table()
+        part = mf._partitions[0][0]
+        at = part.get()
+
+        if part.frame_id is None:
+            part.frame_id = import_table(at)
+
+        alias.append("WITH " if len(alias) == 0 else "\n),\n")
+        alias.extend((name, " AS (\n", "  SELECT\n"))
+
+        for i, col in enumerate(at.column_names):
+            alias.append("    " if i == 0 else ",\n    ")
+            # Cut the "F_" prefix from the column name
+            alias.extend(('"', col, '"', " AS ", '"', col[2:], '"'))
```

Comment:
what if column name contains a double quote?

Reply:
SQL Error is raised

Reply:
So you're sure sql injection won't happen here, right?

Reply:
SQL injection is quite possible here. Do you think we need a special processing for column names to avoid SQL injections in the column names?

Reply:
Some SQL query may insert malicious columns into a table. Can we use @ for parameters (columns)?

Reply:
I'm not owning this backend, it's your call guys. But I personally would rather _not_ introduce a possibility for an injection, maybe raise an error if any column name contains a `"` in it?

Reply:
I think, it should be done in a separate PR and in a different way. Of course, if it's really required.
Currently, all column names are encoded by just adding the "F_" prefix. Probably, this is not reliable and some other encoding should be used, for example, Base62.

Reply:
We have no API in our SQL front end to safely substitute parameters. But both frame (with its column names) and SQL query come from the same person, so what are we actually protecting from? Can you provide us with an example of an attack scenario? SQL injection usually happens through query parameters (and we don't have them), not column names.

We are unlikely to add new SQL features in HDK. So, we can force alphanumeric symbols only in column names (or some weaker filter) to exclude the possibility to put subqueries there.